src/mds/objects.py

```diff
@@ -227,6 +226,65 @@ async def create_object_for_id(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.get("/objects/{guid:path}/download")
+async def get_object_signed_download_url(
+    guid: str,
+    request: Request,
+) -> JSONResponse:
+    """
+    Send a GET request to the data access service to generate a signed download
+    url for the given GUID or alias. Returns the generated signed download url
+    to the user.
+
+    Args:
+        guid (str): indexd GUID or alias
+        request (Request): starlette request (which contains reference to FastAPI app)
+
+    Returns:
+        200: { "guid": GUID or alias, "download_url": signed download url }
+        404: if the data access service can not find GUID/alias in indexd
+        403: if the data access service returns a 401 or a 403
+        500: if there is an error making the request to the data access service
+        or the data access service returns any other 400-range or 500-range
+        error
+    """
+    try:
+        endpoint = (
+            config.DATA_ACCESS_SERVICE_ENDPOINT.rstrip("/") + f"/data/download/{guid}"
+        )
+        auth_header = str(request.headers.get("Authorization", ""))
+        logger.debug(
+            f"Attempting to GET signed download url from data access service for GUID or alias '{guid}'"
+        )
+        response = await request.app.async_client.get(
+            endpoint, headers={"Authorization": auth_header}
```

Comment:
looking at the Fence code, if the ID is an alias, i don't think this would work. you might have to query the `did` first like in the other endpoints

Reply:
Yeah I tested this and Fence is unable to find the record in Indexd since it requests `GET http://indexd-service/index/{alias}` instead of `GET http://indexd-service/{alias}`.

What about alternatively making the change in Fence so that it can generate signed urls for aliases? It might just be a matter of changing `url = indexd_server + "/index/"` to `url = indexd_server + "/"` (https://github.com/uc-cdis/fence/blob/50ff56879f7c2f74442c18e4be77ea389b52d38d/fence/blueprints/data/indexd.py#L291)?

Reply:
that'd be fine with me

Reply:
=> PXP-7007